shared/nav.desktop.js

```diff
@@ -64,51 +65,77 @@ class Nav extends Component {
   }
 
   _resizeWindow () {
-    this._askedResize = false
-    const currentWindow = remote.getCurrentWindow()
-
-    if (!currentWindow) {
+    if (this._askedResize) {
       return
     }
 
-    if (this._renderedActiveTab === loginTab) {
-      const [width, height] = currentWindow.getContentSize()
+    this._askedResize = true
+
+    setImmediate(() => {
+      this._askedResize = false
+      const currentWindow = remote.getCurrentWindow()
 
-      // Did we actually change the size
-      if (width !== globalResizing.login.width && height !== globalResizing.login.height) {
-        this._originalSize = {width, height}
+      if (!currentWindow) {
+        return
       }
 
-      currentWindow.setContentSize(globalResizing.login.width, globalResizing.login.height, true)
-      currentWindow.setResizable(false)
-    } else {
-      if (this._originalSize) {
-        const {width, height} = this._originalSize
-        currentWindow.setContentSize(width, height, true)
-      } else {
-        currentWindow.setContentSize(globalResizing.normal.width, globalResizing.normal.height, true)
+      const activeTab = this._activeTab()
+
+      if (this._lastResizedTab === activeTab) {
+        return
       }
-      currentWindow.setResizable(true)
-    }
+
+      const oldWasLogin = this._lastResizedTab === loginTab
+      const newIsLogin = activeTab === loginTab
+
+      // going to/from login?
+      if (newIsLogin) {
+        const [width, height] = currentWindow.getContentSize()
+
+        // Did we actually change the size
+        if (width !== globalResizing.login.width && height !== globalResizing.login.height) {
+          this._originalSize = {width, height}
+        }
+
+        currentWindow.setContentSize(globalResizing.login.width, globalResizing.login.height, true)
+        currentWindow.setResizable(false)
+        currentWindow.show()
+      } else if (oldWasLogin) {
+        if (!flags.mainWindow) {
```

Comment:
i renamed it so its about checking the tab. where the consequences are to resize/hide